interoperator/internal/renderer/helm/renderer.go

```diff
@@ -159,3 +187,25 @@ func (r *helmRenderer) renderResources(ch *chartapi.Chart, values chartutil.Valu
 		Files: files,
 	}, nil
 }
+
+func (r *helmRenderer) processCRDS(ch *chartapi.Chart) (map[string]string, error) {
+	crds := make(map[string]string)
+	for _, cr := range ch.CRDs() {
+		//Add chart name so that crds are applied first
```

Comment:
Why should CRDs be applied first?

Reply:
The templates can have instances of these CRDs.

Reply:
Ok. But why do they need to be applied first?

Reply:
Helm installs the CRDs first. Then installs the other templates. So I thought we will also do the same. https://helm.sh/docs/topics/charts/#custom-resource-definitions-crds

Since we apply resources serially and retry starting from beginning again on first failure, we will need the order to be correct. If a cr instance is tried to be created before the crd, it will fail and keep on failing on retries because the resources will tried to be created in the same order(cr instance before crd) in the retry also.

Reply:
👍 We should follow the order that helm follows